### PR TITLE
Add global options for performance/precision tradeoffs

### PR DESCRIPTION
**Pull request by xuanluo113**
Pull request overview
---------------------
This object enables users to choose certain options that speed up EnergyPlus simulation, but may lead to small decreases in accuracy of results.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by jasondegraw** [comment]
@XuanLuoLBL I'm not sure if this was intended for the 10x repo, but here it is going to need an NFP.

**Comment by hongtz68** [comment]
> 
> 
> @XuanLuoLBL I'm not sure if this was intended for the 10x repo, but here it is going to need an NFP.

@jasondegraw Jason, this global object is designed and approved by the 10X team (Amir, Mike, Edwin, me etc). No need of an NFP. This object and a few features will go to EnergyPlus 9.2 release. Further questions, let me know.

**Comment by jasondegraw** [comment]
@hongtz68 The NFP provides us with a transparent procedure that documents what is going into the code. That's good for us and for stakeholders.

**Review by lgu1234 — APPROVED** [review]

**Comment by lgu1234** [comment]
@XuanLuoLBL You may need to add description in the Input Output Reference.

File: doc/input-output-reference/src/overview/group-simulation-parameters.tex [hunk 1]
```diff
@@ -790,6 +790,21 @@ \subsubsection{Inputs}\label{inputs-15-014}
...
+\paragraph{Field: Use Coil Direct Solutions}\label{use-coil-direct-solutions}
+
+If Yes, an analytical or empirical solution will be used to replace iterations in the coil performance calculations.
+
+An IDF example:
+
+\begin{lstlisting}
+PerformancePrecisionTradeoffs,
+  Yes;                      !- Use Coil Direct Solutions
+\end{lstlisting}
```
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-simulation-parameters.tex:806]: This documentation does not cover the A2 field `Use Caching In Utility Functions`

File: src/EnergyPlus/SimulationManager.cc [hunk 2]
```diff
@@ -1150,6 +1150,23 @@ namespace SimulationManager {
...
+                                          NumAlpha,
+                                          Number,
+                                          NumNumber,
+                                          IOStat,
+                                          lNumericFieldBlanks,
+                                          lAlphaFieldBlanks,
+                                          cAlphaFieldNames,
+                                          cNumericFieldNames);
+            if (Alphas(1) == "YES") DoCoilDirectSolutions = true;
+        }
```
Comment (Myoldmopar) [inline src/EnergyPlus/SimulationManager.cc:1167]: Should grab the JSON directly...

**Review by Myoldmopar — CHANGES_REQUESTED** [review]

File: doc/input-output-reference/src/overview/group-simulation-parameters.tex (hunk 1 above)
Comment (xuanluo113) [reply doc/input-output-reference/src/overview/group-simulation-parameters.tex:806]: @Myoldmopar Thanks. We decided to get rid of the A2 field for now. Please see the latest commit.

File: src/EnergyPlus/SimulationManager.cc (hunk 2 above)
Comment (xuanluo113) [reply src/EnergyPlus/SimulationManager.cc:1167]: @Myoldmopar I followed the pattern. Can you give a reference example for this? Thanks.

Comment (Myoldmopar) [reply src/EnergyPlus/SimulationManager.cc:1167]:
```c++
auto const instances = inputProcessor->epJSON.find("PerformancePrecisionTradeoffs");
if (instances != inputProcessor->epJSON.end()) {
    auto &instancesValue = instances.value();
    for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
        auto const &fields = instance.value();
        auto const &thisObjectName = UtilityRoutines::MakeUPPERCase(instance.key());
        inputProcessor->markObjectAsUsed(thisObjectType, thisObjectName);
        if (fields.find("do_direct_coil_solution") != fields.end()) {
            if (UtilityRoutines::MakeUPPERCase(fields.at("do_direct_coil_solution")) == "YES") {
                DoCoilDirectSolutions = true;
            }
        }
    }
}
```

File: doc/input-output-reference/src/overview/group-simulation-parameters.tex (hunk 1 above)
Comment (Myoldmopar) [reply doc/input-output-reference/src/overview/group-simulation-parameters.tex:806]: :+1:

**Review by Myoldmopar — APPROVED** [review]

**Pull request merged**